### PR TITLE
feat(okta): list all users

### PR DESCRIPTION
**Pull request by dannylwe**
## Checklist before requesting a review (skip if just adding/editing APIs & templates)
- [x] I added tests, otherwise the reason is:
- [x] External API requests have `retries`
- [x] Pagination is used where appropriate
- [x] The built in `nango.paginate` call is used instead of a `while (true)` loop
- [x] Third party requests are NOT parallelized (this can cause issues with rate limits)
- [ ] If a sync requires metadata the `nango.yaml` has `auto_start: false`
- [ ] If the sync is a `full` sync then `track_deletes: true` is set

### CODE REVIEW

File: integrations/okta/nango.yaml [hunk 1]
```diff
@@ -1,5 +1,14 @@
 integrations:
     okta:
+        syncs:
+            user:
```
Comment (khaliqgant) [inline integrations/okta/nango.yaml:4]: `users`

**Review by khaliqgant — CHANGES_REQUESTED** [review]
Couple of changes requested. Main one is that the sync should be called `users`

Otherwise, looks good and works as expected.

File: integrations/okta/nango.yaml [hunk 2]
```diff
@@ -83,3 +92,5 @@ models:
             secondEmail: string | null
             login: string
             email: string
+    OktaListUsersResponse:
```
[the remark this answers is not in the document]
Comment (dannylwe) [reply integrations/okta/nango.yaml]: resolved

File: integrations/okta/nango.yaml (hunk 1 above)
Comment (dannylwe) [reply integrations/okta/nango.yaml]: resolved

File: integrations/okta/mocks/list-all-users/output.json [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+            "type": {
+                "id": "otyk6uwil7SusQigU697"
+            },
+            "profile": {
+                "firstName": "Isaac",
+                "lastName": "Brock",
+                "mobilePhone": "555-415-1337",
+                "secondEmail": null,
+                "login": "hassanwari254@gmaile.com",
+                "email": "hassanwari254@gmail.com"
```
[the remark this answers is not in the document]
Comment (dannylwe) [reply integrations/okta/mocks/list-all-users/output.json]: resolved

**Review by mungaihosea — APPROVED** [review]

**Review by khaliqgant — APPROVED** [review]

**Pull request merged**